content/en/docs/concepts/scheduling-eviction/pod-eviction.md

```diff
@@ -0,0 +1,375 @@
+| `MemoryPressure`  | `memory.available`                                                                    | Available memory on the node has satisfied an eviction threshold                                                             |
+| `DiskPressure`    | `nodefs.available`, `nodefs.inodesFree`, `imagefs.available`, or `imagefs.inodesFree` | Available disk space and inodes on either the node's root filesystem or image filesystem has satisfied an eviction threshold |
+| `PIDPressure`     | `pid.available`                                                                       | Available processes identifiers on the (Linux) node has fallen below an eviction threshold                                   |
+
+The kubelet updates the node conditions based on the configured 
+`--node-status-update-frequency`, which defaults to `10s`.
+
+### Node condition oscillation
+
+In some cases, nodes oscillate above and below soft eviction thresholds without
+holding for the defined grace periods. This causes the reported node condition
+to constantly switch between `true` and `false`, leading to bad eviction decisions.
+
+To protect against oscillation, you can use the `eviction-pressure-transition-period`
+flag, which controls how long the kubelet must wait before transitioning a node
+condition to a different state.
+
+## Reclaiming node level resources {#reclaim-node-resources}
+
+The kubelet tries to reclaim node-level resources before it evicts end-user pods.
+
+When a `DiskPressure` node condition is reported, the kubelet reclaims node-level
+resources based on the filesystems on the node. 
+
+### With `imagefs`
+
+If the node has a dedicated `imagefs` filesystem for container runtimes to use,
+the kubelet does the following:
+
+  * If the `nodefs` filesystem meets the eviction threshlds, the kubelet deletes
+    dead pods and containers. 
+  * If the `imagefs` filesystem meets the eviction thresholds, the kubelet
+    deletes all unused images. 
+
+### Without `imagefs`
+
+If the node only has a `nodefs` filesystem that meets eviction thresholds,
+the kubelet frees up disk space in the following order:
+
+1. Delete dead pods and containers
+1. Delete unused images
+
+## Evicting end-user pods
+
+If the kubelet's attempts to reclaim node-level resources don't bring the eviction
+signal below the threshold, the kubelet begins to evict end-user pods. 
+
+The kubelet uses the following parameters to determine pod eviction order:
+
+1. Pod resource usage relative to defined requests
+1. [Pod priority](/docs/concepts/configuration/pod-priority-preemption/)
+1. Resource consumption relative to scheduling requests
+1. Pod [QoS class](/docs/tasks/configure-pod-container/quality-service-pod/)
+
+As a result, kubelet ranks and evicts pods in the following order:
```

Comment:
This only applies to CPU and memory, as QoS doesn't take ephemeral storage requests into account.

Reply:
So something like "Note: the QoS class ranking does not apply to ephemeral storage requests like `inodes` and `PIDs`."?

Reply:
I'd suggest framing this section differently, if we keep it at all.  The QoS descriptions are meant to help users reason about behavior, rather than completely describe the algorithm.
```
For example, if a node is under MemoryPressure, it would evict pods in the following order:

1. BestEffort or Burstable pods are the only pods where memory usage is allowed to exceed requests.  Among pods where usage is greater than requests, those with the lowest Priority, and then those using the most in excess of their requests are evicted first.
2. `Guaranteed` pods may not exceed their requests, since their requests and limits are equal.  Burstable pods may also not be exceeding their requests when MemoryPressure is triggered.  These pods are the last to be evicted since they are operating within their requested resources.

Note: QoS is not actually used during ranking, but can help illustrate which pods are most likely to be evicted.  QoS does not take EphemeralStorage requests into account, so the above scenario won't apply if the node is under DiskPressure, for example.
```

Reply:
I like that suggestion a lot, especially based on your other comment about potentially removing QoS at some point. I'll rework.